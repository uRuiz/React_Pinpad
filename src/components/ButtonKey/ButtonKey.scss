.button-key {
  height: 64px;
  width: 64px;
  color: #2d2d2d;
  background-color: #ffc419;
  border: none;
  border-radius: 50%;
  margin-left: 8px;
  margin-right: 8px;
  cursor: pointer;
  font-size: 1.3rem;
}

.button-key:hover {
  border: 1px solid white;
  background-color: #ffd55e;
}

.button-key:active {
  background-color: #e0a700;
}