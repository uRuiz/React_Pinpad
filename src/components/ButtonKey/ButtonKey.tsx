import './ButtonKey.scss';

interface iButtonProps {
  value: string;
  onButtonClick: any;
  disabled: boolean
}

export const ButtonKey = ({value, onButtonClick, disabled}: iButtonProps) => {
  return (
    <button className="button__key" disabled={disabled} onClick={() => onButtonClick(value)}>
        {value}
    </button>
  )
}