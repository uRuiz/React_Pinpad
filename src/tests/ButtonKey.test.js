import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { ButtonKey } from '../components/ButtonKey/ButtonKey';

describe('Tests in <Button />', () => {

  const value = "1";
  const wrapper = shallow(<ButtonKey value={value}/>);

  test('It should show <ButtonKey /> properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show the value and disables props', () => {    
    const disabled = true;
    const wrapper = shallow(
      <ButtonKey 
        value={value}
        disabled={disabled}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  test('It should show the value of the button', () => {
    const valueButton = wrapper.find('button').text()

    expect(valueButton).toBe(value);
  });
})
