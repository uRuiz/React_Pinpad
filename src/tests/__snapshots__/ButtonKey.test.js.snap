// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Button /> It should show <ButtonKey /> properly 1`] = `
<button
  className="button__key"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Tests in <Button /> It should show the value and disables props 1`] = `
<button
  className="button__key"
  disabled={true}
  onClick={[Function]}
>
  1
</button>
`;
